graphql_compiler/compiler/ir_sanity_checks.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2017 Kensho Technologies, Inc.
+"""Functions that ensure the IR generated by the front-end satisfies all invariants."""
```

Comment:
general suggestion to maybe add 

```
def check_assertion(condition, error_message):
    if not condition:
        raise AssertionError(error_message)
```

though I see you have a lot of corner cases, and this is pretty readable, so definitely just a suggestion!

Reply:
Yeah I've thought about it. I think I'll hold off for now, because I'm not a huge fan of passing complex expressions to functions:
```
check_assertion(this is that or 'foo' in my_dict, 'expectations were not confirmed')
```